src/tests/Loader/classloader/StaticVirtualMethods/DiamondShape/svm_diamondshape.cs

```diff
@@ -0,0 +1,264 @@
+        Console.WriteLine("At IFooEx.Foo");
+        return a + 2;
+    }        
+}
+
+class FooClass : IFoo2, IFooEx
+{
+    // Dummy
+    public static int Foo(int a)
+    {
+        return 0;
+    }
+}
+
+interface I1
+{
+    static int Func(int a);
+}
+
+interface I2 : I1
+{
+    // static int I1.Func(int a) { return a + 2; }
+}
+
+interface I3 : I1
+{
+    // static int I1.Func(int a) { return a + 3; }
+}
+
+interface I4 : I2, I3
+{
+    // static int I1.Func(int a) { return a + 4; }
+}
+
+class I4Class : I4
+{
+    // @REMOVE
+    static int I1.Func(int a)
+    {
+        Console.WriteLine("At I4Class.Func");
+        return a + 4;
+    }
+}
+
+interface I5: I1 
+{
+    // static int I1.Func(int a) { return a + 5; }
+}
+
+interface I6: I1 
+{
+    // static int I1.Func(int a) { return a + 6; }
+}
+
+interface I7: I5, I6
+{
+    // static int I1.Func(int a) { return a + 7; }
+}
+
+interface I8: I4, I7
+{
+    // static int I1.Func(int a) { return a + 8; }
+}
+
+class I47Class: I4, I7
+{
+    // @REMOVE
+    static int I1.Func(int a)
+    {
+        Console.WriteLine("At I4Class.Func");
+        return a + 8;
+    }            
+
+}
+
+class I8Class: I8
+{
+    // @REMOVE
+    static int I1.Func(int a)
+    {
+        Console.WriteLine("At I4Class.Func");
+        return a + 8;
+    }            
+}
+
+interface GI1<T>
+{
+    static int Func<S>(out Type[] types); 
+}
+
+interface GI2<T> : GI1<T>
+{
+    // static int GI1<T>.Func<S>(out Type[] types) { Console.WriteLine(typeof(T) + ", "typeof(S) + ", GI1Class"); types = new Type[] { typeof(T), typeof(S) }; return 2; }  
+
+} 
+
+interface GI3<T> : GI1<T>
+{
+    // static int GI1<T>.Func<S>(out Type[] types) { Console.WriteLine(typeof(T) + ", "typeof(S) + ", GI1Class"); types = new Type[] { typeof(T), typeof(S) }; return 3; }  
+} 
+
+interface GI4<T> : GI2<T>, GI3<T>
+{
+    // static int GI1<T>.Func<S>(out Type[] types) { Console.WriteLine(typeof(T) + ", "typeof(S) + ", GI1Class"); types = new Type[] { typeof(T), typeof(S) }; return 4; }  
+} 
+
+class GI23Class<T>: GI2<T>, GI3<T>
+{
+    // @REMOVE
+    static int GI1<T>.Func<S>(out Type[] types) { Console.WriteLine(typeof(T) + ", " + typeof(S) + ", GI1Class"); types = new Type[] { typeof(T), typeof(S) }; return 4; }  
+}
+
+class GI4Class<T>: GI4<T>
+{
+    // @REMOVE
+    static int GI1<T>.Func<S>(out Type[] types) { Console.WriteLine(typeof(T) + ", " + typeof(S) + ", GI1Class"); types = new Type[] { typeof(T), typeof(S) }; return 4; }  
+}
+
+class Program
+{
+    private static void CallFoo<T>(int value)
+        : T is IFoo
+    {
+        T.Foo(value);
+    }
+    
+    private static void CallI1Func<T>(int value)
+        : T is I1
+    {
+        T.Func(value);
+    }
+    
+    private static void CallGI1Func<T, U, V>(out Type[] types)
+        : T is GI1<U>
+    {
+        T.Func<V>(out types);
+    }
+    
+    public static void Negative()
+    {
+        Console.WriteLine("Calling IFoo.Foo on Foo - expecting exception.");
```

Comment:
We'll also need positive/negative cases for scenarios involving function pointers.

Reply:
I have copy & pasted and curated each of the test scenarios to exercise call / ldftn in the latest commit.